pandas/core/algorithms.py

```diff
@@ -304,17 +304,18 @@ def _value_counts_arraylike(values, dropna=True):
 
     orig = values
 
+    if is_period:
+        from pandas.tseries.period import PeriodIndex
+        index = PeriodIndex(values)
+        freq = index.freq
+        values = index.values
+
     from pandas.core.series import Series
     values = Series(values).values
```

Comment:
maybe reorg this a bit, we should catch the datetime/timedelta/period ones before we do these coercion lines